### ISSUE TITLE
Inconsistent shape of index returned from F.max_pool2d_with_indices

### ISSUE THREAD

**Issue by phizaz** [pytorch/pytorch#39326]
## 🐛 Bug

Inconsistent shape of index returned from `F.max_pool2d_with_indices` when used with CUDA tensor. This applies to 3-dimensional tensors.

## To Reproduce

```
x = torch.randn(2, 2, 2).to('cuda')
pred, idx = F.max_pool2d_with_indices(x, 2)
print(pred.shape) # torch.Size([2, 1, 1])
print(idx.shape) # torch.Size([1, 2, 1, 1]) which is inconsistent
```

Interestingly, this doesn't happen to CPU's tensor.

```
x = torch.randn(2, 2, 2).to('cpu')
pred, idx = F.max_pool2d_with_indices(x, 2)
print(pred.shape) # torch.Size([2, 1, 1])
print(idx.shape) # torch.Size([2, 1, 1]), expected
```

## Environment

```
PyTorch version: 1.5.0
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Ubuntu 18.04.4 LTS
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
CMake version: Could not collect

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.1.105
GPU models and configuration: GPU 0: GeForce GTX 1060 6GB
Nvidia driver version: 440.44
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] efficientnet-pytorch==0.6.3
[pip] msgpack-numpy==0.4.4.3
[pip] numpy==1.18.1
[pip] segmentation-models-pytorch==0.1.0
[pip] torch==1.5.0
[pip] torchtext==0.5.0
[pip] torchvision==0.6.0a0+82fd1c8
[conda] blas                      1.0                         mkl
[conda] efficientnet-pytorch      0.6.3                    pypi_0    pypi
[conda] mkl                       2020.0                      166
[conda] mkl-service               2.3.0            py37he904b0f_0
[conda] mkl_fft                   1.0.15           py37ha843d7b_0
[conda] mkl_random                1.1.0            py37hd6b4f25_0
[conda] pytorch                   1.5.0           py3.7_cuda10.1.243_cudnn7.6.3_0    pytorch
[conda] segmentation-models-pytorch 0.1.0                     dev_0    <develop>
[conda] torchtext                 0.5.0                    pypi_0    pypi
[conda] torchvision               0.6.0                py37_cu101    pytorch
```

**Comment by xwang233** [comment]
Thanks for the report! The same issue has been posted a few days ago at #38986, and fixed at #38992. Please check the nightly build.

**Comment by izdeby** [comment]
closing as a duplicate
